torch/_inductor/memory.py

```diff
@@ -427,16 +430,30 @@ class BufferInfo(TypedDict):
     # schedule nodes one at a time
     schedule: List[BaseSchedulerNode] = []
     num_iters: int = 0
+    size_threshold = percentile([node.mpi_node.size for node in nodes], 95)
     while num_iters < len(nodes) and nodes_to_schedule:
         # select a node to schedule:
-        selected_node = min(
-            nodes_to_schedule,
-            key=lambda node: (
-                max(live_memory + node.mpi_node.size, max_memory),
-                node.mpi_node.size - node_info[node]["memory_to_free"],
-                node.mpi_node.index,
-            ),
-        )
+        if min(node.mpi_node.size for node in nodes_to_schedule) > size_threshold:
+            selected_node = min(
+                nodes_to_schedule,
+                key=lambda node: min(
+                    (
+                        succ_node.mpi_node.index
+                        for succ_node in node.mpi_node.succ_nodes
+                    ),
+                    default=len(nodes),
+                ),
+            )
+        else:
+            selected_node = min(
+                nodes_to_schedule,
+                key=lambda node: (
+                    node.mpi_node.size if node.mpi_node.size > memory_gap else 0,
```

Comment:
It is significantly faster to do it this way with `memory_gap` than the original way, though I am not sure why.